core/modules/path/path.module

```diff
@@ -453,10 +808,875 @@ function path_entity_update(Entity $entity) {
+  $entity_paths = &backdrop_static(__FUNCTION__);
+
+  if (!isset($entity_paths)) {
+    $entity_paths = array();
+    foreach (entity_get_info() as $entity_type => $entity_info) {
+      if (isset($entity_info['default path'])) {
+        $default_path = $entity_info['default path'];
+        $default_path = preg_quote($default_path, '/');
+        $default_path = str_replace(preg_quote('%' . $entity_type, '/'), '(\d+)', $default_path);
+        $entity_paths[$entity_type] = $default_path;
+      }
+    }
+  }
+
+  foreach ($entity_paths as $entity_type => $default_path) {
+    if (preg_match("/^{$default_path}$/", $path, $matches)) {
+      if ($entity = entity_load($entity_type, array($matches[1]))) {
+        return array('entity_type' => $entity_type, 'entity' => reset($entity));
+      }
+      break;
+    }
+  }
+}
+
+/**
+ * Check the ability to perform redirects with the current request context.
+ *
+ * This function checks the following conditions:
+ * - If the PHP entry point is the root index.php file.
+ * - If PHP is not running as CLI.
+ * - If the site is not offline or in install/update mode.
+ * - If the curerent page is not an admin page (check can be disabled).
+ * - If the current request does not have any POST data since a redirect
+ *   may interrupt form submission.
+ *
+ * @return
+ *   TRUE if redirections can be performed, or FALSE otherwise.
+ */
+function redirect_can_redirect() {
+  $can_redirect = &backdrop_static(__FUNCTION__);
+  $config = config('redirect.settings');
+  if (!isset($can_redirect)) {
+    $path = current_path();
+    $can_redirect = TRUE;
+
+    if ($_SERVER['SCRIPT_NAME'] != $GLOBALS['base_path'] . 'index.php') {
+      // Do not redirect if the root script is not /index.php.
+      $can_redirect = FALSE;
+    }
+    elseif (!empty($_POST)) {
+      // Do not redirect if this is a post request with data.
+      $can_redirect = FALSE;
+    }
+    elseif (backdrop_is_cli()) {
+      // If this is a command line request (Drush, etc), skip processing.
+      $can_redirect = FALSE;
+    }
+    elseif ((state_get('maintenance_mode', 0) || defined('MAINTENANCE_MODE')) && !user_access('access site in maintenance mode')) {
+      // Do not redirect in offline or maintenance mode.
+      $can_redirect = FALSE;
+    }
+    elseif (!$config->get('global_admin_paths') && path_is_admin($path)) {
+      // Do not redirect on admin paths.
+      $can_redirect = FALSE;
+    }
+  }
+
+  return $can_redirect;
+}
+
+/**
+ * Compare that all values and associations in one array match another array.
+ *
+ * We cannot use array_diff_assoc() here because we need to be recursive.
+ *
+ * @param $match
+ *   The array that has the values.
+ * @param $haystack
+ *   The array that will be searched for values.
+ * @return
+ *   TRUE if all the elements of $match were found in $haystack, or FALSE
+ *   otherwise.
+ */
+function redirect_compare_array_recursive($match, $haystack) {
```

Comment:
I think instead of adding this function, we may be able to use core's `backdrop_array_diff_assoc_recursive()`.